lint/clang_tidy.bzl

```diff
@@ -369,7 +371,7 @@ def _clang_tidy_aspect_impl(target, ctx):
     if not CcInfo in target:
         return []
 
-    files_to_lint = _filter_srcs(ctx.rule)
+    files_to_lint = _filter_srcs(ctx.rule, ctx.attr._skip_if_no_srcs)
     compilation_context = target[CcInfo].compilation_context
```

Comment:
Now I look at it - compilation_context could be None - see https://bazel.build/rules/lib/providers/CcInfo#CcInfo; I suspect that this will happen precisely when there are no srcs present and no deps? And if so we might need to protect other places further down where we access compilation_context?

Reply:
@johan-vcc does it fix the issue for you if you check for this being `None` ?

Reply:
Tried it out, then that check would need to be before it tries to access rule.files.srcs in _filter_srcs().
However, in my case the targets have compilation_context so that would not avoid the need to check srcs as well.